sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -411,6 +415,105 @@ case class GpuContains(left: Expression, right: Expression)
       doColumnar(expandedLhs, rhs)
     }
   }
+
+  /**
+   * Get a combiner that can be used to find candidates to combine
+   */
+  override def getCombiner(): GpuExpressionCombiner = new ContainsCombiner(this)
+}
+
+case class GpuMultiContains(left: Expression, targets: Seq[UTF8String], output: StructType)
+    extends GpuExpression with ShimExpression {
+
+  override def otherCopyArgs: Seq[AnyRef] = Nil
+
+  override def dataType: DataType = output
+
+  override def nullable: Boolean = false
+
+  override def prettyName: String = "multi_contains"
+
+  override def columnarEval(batch: ColumnarBatch): GpuColumnVector = {
+    val targetsBytes = targets.map(t => t.getBytes).toArray
+    withResource(ColumnVector.fromUTF8Strings(targetsBytes: _*)) { targetsCv =>
+      withResource(left.columnarEval(batch)) { lhs =>
+        withResource(lhs.getBase.stringContains(targetsCv)) { boolCvs =>
+          val retView = ColumnView.makeStructView(batch.numRows(), boolCvs: _*)
+          GpuColumnVector.from(retView.copyToColumnVector(), dataType)
+        }
+      }
+    }
```

Comment:
To reduce the peak GPU memory usage.
```suggestion
    val boolCvs = withResource(ColumnVector.fromUTF8Strings(targetsBytes: _*)) { targetsCv =>
      withResource(left.columnarEval(batch)) { lhs =>
        lhs.getBase.stringContains(targetsCv)
      }
    }
    withResource(boolCvs) { _ =>
      val retView = ColumnView.makeStructView(batch.numRows(), boolCvs: _*)
      GpuColumnVector.from(retView.copyToColumnVector(), dataType)
    }
```

Reply:
`boolCvs` contains the main result/memory, they are returned and needed. Closing boolCvs will not release the main memory, because the `GpuColumnVector.from` will increase the reference count.

Reply:
I mean we can close `targetsCv ` and `lhs` before creating the result struct column. Since the result column only depends on the `boolCvs`.
The original code will hold `targetsCv `, `lhs` and `boolCvs` at the same time, taking more GPU memory when creating the result column.